### ISSUE TITLE
While installing postgresql@11: configure: error: header file <perl.h> is required for Perl

### ISSUE THREAD

**Issue by hajhatten** [petere/homebrew-postgresql#44]
```
==> ./configure --prefix=/usr/local/Cellar/postgresql@11/11.0 --enable-dtrace --enable-nls --with-bonjour --with-gssa
Last 15 lines from /Users/hajhatten/Library/Logs/Homebrew/postgresql@11/01.configure:
checking for library containing bind_textdomain_codeset... -lintl
checking libintl.h usability... yes
checking libintl.h presence... yes
checking for libintl.h... yes
checking for msgfmt... /usr/local/opt/gettext/bin/msgfmt
checking for msgfmt flags... -c
checking for msgmerge... /usr/local/opt/gettext/bin/msgmerge
checking for xgettext... /usr/local/opt/gettext/bin/xgettext
checking for tclsh... /usr/local/opt/tcl-tk/bin/tclsh
checking for tclConfig.sh... /usr/local/Cellar/tcl-tk/8.6.8/lib/tclConfig.sh
checking tcl.h usability... yes
checking tcl.h presence... yes
checking for tcl.h... yes
checking for perl.h... no
configure: error: header file <perl.h> is required for Perl

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/petere/homebrew-postgresql/issues

These open issues may also help:
Fully qualified name petere/postgresql/postgresql@10 is needed to install correct formula https://github.com/petere/homebrew-postgresql/issues/40
```

OSX Mojave 10.14 (18A391)
Homebrew 1.7.7
Homebrew/homebrew-core (git revision 5e88c5; last commit 2018-10-23)
Homebrew/homebrew-cask (git revision 5c3b4; last commit 2018-10-23)

Earlier had 10.5 installed, but ran:
```
brew uninstall postgresql --force
brew tap petere/postgresql
brew install postgresql@11
```

**Comment by simonmichael** [comment]
Same issue here, with postgresql@9.3.

**Comment by simonmichael** [comment]
Possible duplicate of https://github.com/petere/homebrew-postgresql/issues/41, which has a [workaround](https://github.com/petere/homebrew-postgresql/issues/41#issuecomment-405048183).

**Comment by dcnorris** [comment]
This simple [workaround](https://github.com/petere/homebrew-postgresql/issues/41#issuecomment-442178698) worked nicely, without need for `brew edit`.

**Comment by austinmcconnell** [comment]
This is what finally worked for me. Open the folder `/Library/Developer/CommandLineTools/Packages/` and manually install `macOS_SDK_headers_for_macOS_10.14.pkg`. Postgres install worked fine after that.

**Comment by black-snow** [comment]
Hitting this now with 11.2_1 -> 11.3

> checking for perl.h... no
> configure: error: header file <perl.h> is required for Perl

**Comment by townge** [comment]
Still receiving this error in Catalina. I was able to use the workarounds in Mojave but now the directory to reinstall CLT is gone.

**Comment by ckcollab** [comment]
This was fixed for me on Catalina just by installing XCode via the app store, phew!

**Comment by Carolyn95** [comment]
> Still receiving this error in Catalina. Reinstalling CLTools doesn't work because `/Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg` no longer exists in Catalina so there seems to be no current workaround.
> 
> **Update**: the `brew edit` workaround does not work with postgresql `@11` or `@12`. Fortunately, I was able to get postgresql 11.5 installed successfully from brew by downloading [Xcode 11.2 Beta](https://developer.apple.com/download/) placing it in my _/Applications_ folder, and running (as root) `sudo xcode-select --switch /Applications/Xcode-beta.app/` After that, I could `brew install postgresql` without any Perl errors.

You are right, Catalina does not have the header file, but my system version is 10.15, aka, it doesn't have `/Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.15.pkg`

After installing XCode from App Store, the error was still there, however, running `sudo xcode-select --switch /Applications/Xcode.app/` helped.

Thanks a lot, now I can brew install postgresql smoothly!!
